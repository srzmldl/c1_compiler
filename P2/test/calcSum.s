	.file	"calcSum.c"
	.section	.rodata
.LC0:
	.string	"%d"
.LC1:
	.string	" + %d"
.LC2:
	.string	"= %d\n"
	.text
	.globl	main
	.type	main, @function
main:
.LFB0:
	.cfi_startproc
	pushq	%rbp
	.cfi_def_cfa_offset 16
	.cfi_offset 6, -16
	movq	%rsp, %rbp
	.cfi_def_cfa_register 6
	subq	$64, %rsp
	movl	$10, -52(%rbp)
	movl	$10, -48(%rbp)
	movl	$7, -44(%rbp)
	movl	$5, -40(%rbp)
	movl	$100, -36(%rbp)
	movl	$1, -32(%rbp)
	movl	$7, -28(%rbp)
	movl	$67, -24(%rbp)
	movl	$4, -20(%rbp)
	movl	$99, -16(%rbp)
	movl	$6, -12(%rbp)
	movl	$0, -60(%rbp)
	movl	$0, -56(%rbp)
	jmp	.L2
.L3:
	movl	-60(%rbp), %eax
	cltq
	movl	-48(%rbp,%rax,4), %eax
	addl	%eax, -56(%rbp)
	addl	$1, -60(%rbp)
.L2:
	movl	-60(%rbp), %eax
	cmpl	-52(%rbp), %eax
	jl	.L3
	movl	-48(%rbp), %eax
	movl	%eax, %esi
	movl	$.LC0, %edi
	movl	$0, %eax
	call	printf
	movl	$0, -60(%rbp)
	jmp	.L4
.L5:
	movl	-60(%rbp), %eax
	cltq
	movl	-48(%rbp,%rax,4), %eax
	movl	%eax, %esi
	movl	$.LC1, %edi
	movl	$0, %eax
	call	printf
	addl	$1, -60(%rbp)
.L4:
	movl	-60(%rbp), %eax
	cmpl	-52(%rbp), %eax
	jl	.L5
	movl	-56(%rbp), %eax
	movl	%eax, %esi
	movl	$.LC2, %edi
	movl	$0, %eax
	call	printf
	leave
	.cfi_def_cfa 7, 8
	ret
	.cfi_endproc
.LFE0:
	.size	main, .-main
	.ident	"GCC: (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4"
	.section	.note.GNU-stack,"",@progbits
