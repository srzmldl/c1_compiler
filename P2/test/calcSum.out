Illegal character< ident, include >< comp, < >< ident, stdio >Illegal character< ident, h >< comp, > >


< void >< ident, main >< parenth, ( >< parenth, ) >
< brace, { >

< const >< int >< ident, n >< assign >< number, 10 >< semicolon >
< const >< int >< ident, a >< bracket, [ >< number, 10 >< bracket, ] >< assign >< brace, { >< number, 10 >< comma >< number, 7 >< comma >< number, 5 >< comma >< number, 100 >< comma >< number, 1 >< comma >< number, 7 >< comma >< number, 67 >< comma >< number, 4 >< comma >< number, 99 >< comma >< number, 6 >< brace, } >< semicolon >
< int >< ident, i >< assign >< number, 0 >< semicolon >
< int >< ident, sum >< assign >< number, 0 >< semicolon >
< while >< parenth, ( >< ident, i >< comp, < >< ident, n >< parenth, ) >
< brace, { >
< ident, sum >< assign >< ident, sum >< op, + >< ident, a >< bracket, [ >< ident, i >< bracket, ] >< semicolon >
< ident, i >< assign >< ident, i >< op, + >< number, 1 >< semicolon >
< brace, } >

< ident, printf >< parenth, ( >Illegal character< op, % >< ident, d >Illegal character< comma >< ident, a >< bracket, [ >< number, 0 >< bracket, ] >< parenth, ) >< semicolon >
< ident, i >< assign >< number, 0 >< semicolon >
< while >< parenth, ( >< ident, i >< comp, < >< ident, n >< parenth, ) >
< brace, { >
< ident, printf >< parenth, ( >Illegal character< op, + >< op, % >< ident, d >Illegal character< comma >< ident, a >< bracket, [ >< ident, i >< bracket, ] >< parenth, ) >< semicolon >
< ident, i >< assign >< ident, i >< op, + >< number, 1 >< semicolon >
< brace, } >
< ident, printf >< parenth, ( >Illegal character< assign >< op, % >< ident, d >Illegal character< ident, n >Illegal character< comma >< ident, sum >< parenth, ) >< semicolon >

< brace, } >